new-client/src/models/Click.js

```diff
@@ -275,7 +304,26 @@ export function handleClick(evt, map, callback) {
                   })
               );
               break;
-            case "text/xml":
+            case "text/xml": {
+              featurePromises.push(
+                response.value.requestResponse
+                  .text()
+                  .then((text) => {
+                    if (text !== undefined && text) {
+                      features.push(
+                        ...getFeaturesFromXml(response.value, text)
+                      );
+                    }
+                  })
+                  .catch((err) => {
+                    console.error(
+                      "GetFeatureInfo couldn't retrieve correct data for the clicked object.",
+                      err
+                    );
+                  })
+              );
+              break;
+            }
```

Comment:
Also, this part looks very much like the next that follows, the only difference being which parser is used depending on XML or GML, i.e. lines 314 and 332. The rest is just the same so consider refactoring.